src/activity_actor.cpp

```diff
@@ -518,48 +508,37 @@ bool aim_activity_actor::load_RAS_weapon()
         // Menu canceled
         return false;
     }
-    int reload_time = 0;
-    reload_time += opt.moves();
-    if( !gun->reload( you, std::move( opt.ammo ), 1 ) ) {
-        // Reload not allowed
-        return false;
-    }
 
     // Burn 0.6% max base stamina without cardio/BMI factored in x the strength required to fire.
-    you.burn_energy_arms( gun->get_min_str() * static_cast<int>( 0.006f *
+    you.burn_energy_arms( - gun->get_min_str() * static_cast<int>( 0.006f *
                           get_option<int>( "PLAYER_MAX_STAMINA_BASE" ) ) );
-    // At low stamina levels, firing starts getting slow.
-    int sta_percent = ( 100 * you.get_stamina() ) / you.get_stamina_max();
-    reload_time += ( sta_percent < 25 ) ? ( ( 25 - sta_percent ) * 2 ) : 0;
 
-    you.mod_moves( -reload_time );
-    loaded_RAS_weapon = true;
+    reload_loc = opt.ammo;
     return true;
 }
 
 void aim_activity_actor::unload_RAS_weapon()
 {
-    // Unload reload-and-shoot weapons to avoid leaving bows pre-loaded with arrows
     avatar &you = get_avatar();
     item_location weapon = get_weapon();
-    if( !weapon || !loaded_RAS_weapon ) {
+    if( !weapon ) {
         return;
     }
 
     gun_mode gun = weapon->gun_current_mode();
     if( gun->has_flag( flag_RELOAD_AND_SHOOT ) ) {
-        int moves_before_unload = you.get_moves();
-
-        // Note: this code works only for avatar
-        item_location loc = item_location( you, gun.target );
-        you.unload( loc, true );
-
-        // Give back time for unloading as essentially nothing has been done.
-        if( first_turn ) {
-            you.set_moves( moves_before_unload );
+        if( !loaded_RAS_weapon ) {
+            // Refund stamina cost if not loaded.
```

Comment:
Don't refund, just only do the cost on reload?

Reply:
The old feature intended to refund moves never worked. In my reverted PR, PC will not reload the bow until pressing `f`ire, and quitting the aim ui also cost no moves. In this PR, PC will not reload the bow until pressing `f`ire or `.` to make aim steadier(codes your reviewed below is for this feature) , so quitting the aim ui in the "first turn" will not cost moves.

Reply:
I missed that, apologies. I don't think we need a refund mechanic though, we should just only burn things when they need to be burnt.

Reply:
Yes, but when writing this PR, I didn't come up with a good replacement to the current plan where stamina needed for reloading the bow is burnt in aim_activity_actor::load_RAS_weapon(). Do you have any idea?
By the way, stamina is also cost in mod_stamina_archery, see https://github.com/CleverRaven/Cataclysm-DDA/issues/73287.

Reply:
```diff
diff --git a/src/activity_actor.cpp b/src/activity_actor.cpp
index 54e7382bdf..6711873e25 100644
--- a/src/activity_actor.cpp
+++ b/src/activity_actor.cpp
@@ -509,9 +509,6 @@ bool aim_activity_actor::load_RAS_weapon()
         return false;
     }

-    // Burn 0.6% max base stamina without cardio/BMI factored in x the strength required to fire.
-    you.burn_energy_arms( - gun->get_min_str() * static_cast<int>( 0.006f *
-                          get_option<int>( "PLAYER_MAX_STAMINA_BASE" ) ) );

     reload_loc = opt.ammo;
     return true;
@@ -527,12 +524,6 @@ void aim_activity_actor::unload_RAS_weapon()

     gun_mode gun = weapon->gun_current_mode();
     if( gun->has_flag( flag_RELOAD_AND_SHOOT ) ) {
-        if( !loaded_RAS_weapon ) {
-            // Refund stamina cost if not loaded.
-            you.burn_energy_arms( gun->get_min_str() * static_cast<int>( 0.006f *
-                                  get_option<int>( "PLAYER_MAX_STAMINA_BASE" ) ) );
-            return;
-        }
         if( gun->ammo_remaining() ) {
             item_location loc = item_location( you, gun.target );
             you.unload( loc, true );
diff --git a/src/ranged.cpp b/src/ranged.cpp
index 2bd9e5ef9a..1b360612da 100644
--- a/src/ranged.cpp
+++ b/src/ranged.cpp
@@ -880,7 +880,7 @@ int Character::fire_gun( const tripoint &target, int shots, item &gun, item_loca
         return 0;
     }
     if( gun.ammo_required() > 0 && !gun.ammo_remaining() && !ammo ) {
-        debugmsg( "%s's gun %s is empty and has no ammo for reloading.", gun.tname() );
+        debugmsg( "%s is empty and has no ammo for reloading.", gun.tname() );
         return 0;
     }
     bool is_mech_weapon = false;
@@ -931,7 +931,10 @@ int Character::fire_gun( const tripoint &target, int shots, item &gun, item_loca
     int delay = 0; // delayed recoil that has yet to be applied
     while( curshot != shots ) {
         if( !!ammo && !gun.ammo_remaining() ) {
-            gun.reload( get_avatar(), ammo, 1 );
+            Character &you = get_avatar();
+            gun.reload( you, ammo, 1 );
+            you.burn_energy_arms( - gun.get_min_str() * static_cast<int>( 0.006f *
+                                  get_option<int>( "PLAYER_MAX_STAMINA_BASE" ) ) );
         }
         if( gun.faults.count( fault_gun_chamber_spent ) && curshot == 0 ) {
             mod_moves( -get_speed() * 0.5 );
@@ -2690,10 +2693,13 @@ target_handler::trajectory target_ui::run()
                 break;
             }

-            if( you->get_wielded_item()->has_flag( flag_RELOAD_AND_SHOOT ) ) {
-                if( !you->get_wielded_item()->ammo_remaining() && activity->reload_loc ) {
+            item_location weapon = activity->get_weapon();
+            if( weapon->has_flag( flag_RELOAD_AND_SHOOT ) ) {
+                if( !weapon->ammo_remaining() && activity->reload_loc ) {
                     you->mod_moves( -RAS_time );
-                    you->get_wielded_item()->reload( get_avatar(), activity->reload_loc, 1 );
+                    weapon->reload( get_avatar(), activity->reload_loc, 1 );
+                    you->burn_energy_arms( - weapon->get_min_str() * static_cast<int>( 0.006f *
+                                           get_option<int>( "PLAYER_MAX_STAMINA_BASE" ) ) );
                     activity->reload_loc = item_location();
                     activity->loaded_RAS_weapon = true;
                 }
```
This could be a solution?

Reply:
That looks good, yes.